Feat: Sharing AMIs from a central account and creating devops role on deployment

manikandan-thangavelu-rl opened the pull request:
Issue #, if available:

Description of changes:

Sharing AMIs from a central account and creating devops role on deployment

Checklist:

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

- [x] Have you successfully deployed to an AWS account with your changes?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully tested with your changes locally?

<!-- For major releases please provide internal ticket id -->

AS review ticket id:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

narendran1560 committed:
SWB-TRE Feature 2 and Feature 3

narendran1560 committed:
SWB-TRE Feature 2 and Feature 3

narendran1560 committed:
Resolved the eslint issue

narendran1560 committed:
Resolved the eslint issue

narendran1560 committed:
Resolved the JS-Lint issues

narendran1560 committed:
Resolved the JS-Lint issues

SanketD92 committed:
chore: vulnerable dependency version fix (#976)

* chore: fix ansi-regex version in lock

narendran1560 committed:
Feature 2 and 3 code review changes added

narendran1560 committed:
Feature 2 and 3 code review changes added

narendran1560 committed:
Merge remote-tracking branch 'upstream/feat-tre-enhancements' into feat-feature-2and3

narendran1560 committed:
resolved the es-lint issues

narendran1560 committed:
resolved the es-lint issues

manikandan-thangavelu-rl committed:
feat: Create Devops role in central account and share the image from devops account

manikandan-thangavelu-rl committed:
Removed unwanted conditions

manikandan-thangavelu-rl committed:
Merge branch 'feat-tre-enhancements' into topic-amisharing

manikandan-thangavelu-rl committed:
Updated package.json and pnpm-lock

narendran1560 committed:
resolved the eslint issue

narendran1560 committed:
resolved the eslint issue

narendran1560 committed:
Merge pull request #64 from narendran1560/feat-feature-2and3

SWB-TRE Feature 2 and Feature 3

manikandan-thangavelu-rl committed:
Updated pnpm lock

manikandan-thangavelu-rl committed:
Updated pnpm-lock with only prep-devops-account changes

manikandan-thangavelu-rl committed:
Updated @aws-ee to @amzn

manikandan-thangavelu-rl committed:
Merge pull request #2 from manikandan-thangavelu-rl/topic-amisharing

feat: Create Devops role in central account and share the image from …

narendran1560 committed:
Merge pull request #67 from manikandan-thangavelu-rl/feat-tre-enhancements

Feat: Sharing AMIs from a central account and creating devops role on deployment

manikandan-thangavelu-rl committed:
Updated Devops Role permissions

manikandan-thangavelu-rl committed:
Updated Devops Role permissions

narendran1560 committed:
Added the feature flag explaination in the example.yml file

narendran1560 committed:
Merge pull request #70 from narendran1560/feat-feature-2and3

Added the feature flag explaination in the example.yml file

narendran1560 committed:
Merge pull request #71 from RLOpenCatalyst/feat-feature-2and3

SWB-TRE Feature 2 and Feature 3

manikandan-thangavelu-rl committed:
Merge remote-tracking branch 'upstream/feat-tre-enchancement-feature-1' into feat-tre-enchancement-feature-1

manikandan-thangavelu-rl committed:
feat: Creating packer images in Devops account when AMI sharing is enabled

manikandan-thangavelu-rl committed:
feat: Creating packer images in Devops account when AMI sharing is enabled

manikandan-thangavelu-rl committed:
feat: Share appstream images with member account from devops account

manikandan-thangavelu-rl committed:
feat: Share appstream images with member account from devops account

SanketD92 commented on main/solution/backend/config/settings/.defaults.yml:
```diff
@@ -289,3 +289,12 @@ egressStoreKmsPolicyWorkspaceSid: Allow workspace environments to access egress
+prepDevopsStackName: prep-devops-account-${self:custom.settings.awsAccountInfo.awsAccountId}-${self:custom.settings.namespace}
+
+# ExternalID to assume the above role
+devopsRoleExternalId: ${self:custom.settings.awsAccountInfo.awsAccountId}
```
External ID is usually an optional string value associated to the account. Why are we passing the account ID here?

manikandan-thangavelu-rl replied:
External ID is used as an additional identifier when another AWS account is trying to assume an IAM role. We are using the base account's account ID as the identifier. But that can be changed to any string if needed.

SanketD92 commented on addons/addon-base-raas/packages/base-raas-services/lib/environment/environment-ami-service.js:
```diff
@@ -102,6 +110,37 @@ class EnvironmentAmiService extends Service {
+    };
+  }
+
+  async checkIfAmiSharingEnabled() {
```
No need for `async` here.

manikandan-thangavelu-rl replied:
Updated PR with the changes

SanketD92 commented on addons/addon-base-raas/packages/base-raas-services/lib/environment/environment-ami-service.js:
```diff
@@ -102,6 +110,37 @@ class EnvironmentAmiService extends Service {
+    return ec2Client;
+  }
+
+  async getDevopsAccountDetails() {
```
No need for `async` here.

manikandan-thangavelu-rl replied:
Updated PR with the changes

SanketD92 commented on addons/addon-base-raas/packages/base-raas-services/lib/environment/environment-ami-service.js:
```diff
@@ -102,6 +110,37 @@ class EnvironmentAmiService extends Service {
+
+  async getEc2Sdk() {
+    const [aws] = await this.service(['aws']);
+    const isAmiSharingEnabled = await this.checkIfAmiSharingEnabled();
```
No need for `await` here.

manikandan-thangavelu-rl replied:
Updated PR with the changes

SanketD92 commented on addons/addon-base-raas/packages/base-raas-services/lib/environment/environment-ami-service.js:
```diff
@@ -102,6 +110,37 @@ class EnvironmentAmiService extends Service {
+    // Get Devops account client if AMI sharing enabled.
+    if (isAmiSharingEnabled) {
+      this.log.info(`AMI Sharing enabled. Reading SDK using DevOps account role`);
+      const { roleArn, externalId } = await this.getDevopsAccountDetails();
```
No need for `await` here.

manikandan-thangavelu-rl replied:
Updated PR with the changes

SanketD92 commented on main/solution/backend/config/infra/cloudformation.yml:
```diff
@@ -770,6 +770,7 @@ Resources:
               - 'arn:aws:iam::*:role/*-cross-account-role' # This is required since when users onboard the main account as a member account, the onboard CFN stack would also reside there
               - 'arn:aws:iam::*:role/*-xacc-env-mgmt'
               - !Sub 'arn:aws:iam::${AWS::AccountId}:role/${self:custom.settings.namespace}-prep-raas-master-MasterRole-*'
+              - 'arn:aws:iam::*:role/${self:custom.settings.globalNamespace}-DevopsRole'
```
Can we add this resource value only if `enableAmiSharing` is set to `true`? [Example](https://github.com/awslabs/service-workbench-on-aws/blob/c6dd022200fde668476f5aee247f2383d8d54bf2/main/solution/backend/config/infra/cloudformation.yml#L3)

narendran1560 committed:
Resolved aws-swb PR comments

dependabot[bot] committed:
chore(deps): bump cross-fetch from 3.1.4 to 3.1.5 in /docs (#982)

* chore(deps): bump cross-fetch from 3.1.4 to 3.1.5 in /docs

Bumps [cross-fetch](https://github.com/lquixada/cross-fetch) from 3.1.4 to 3.1.5.
- [Release notes](https://github.com/lquixada/cross-fetch/releases)
- [Commits](https://github.com/lquixada/cross-fetch/compare/v3.1.4...v3.1.5)

---
updated-dependencies:
- dependency-name: cross-fetch
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

* chore: trigger build

* fix: tag pnpm version for github pipeline

* fix: use same pnpm tagged version in pipelines

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Sanket Dharwadkar <sdharwad@amazon.com>

SanketD92 committed:
fix: create log group for flow logs (#984)

rsmayda commented on addons/addon-base-raas/packages/base-raas-cfn-templates/src/templates/onboard-account.cfn.yml:
```diff
@@ -106,6 +106,16 @@ Parameters:
+  EnableAmiSharing:
+    Type: String
+    AllowedValues: [true, false]
+    Description: Use appstream image from the central devops account
```
could you add a `Default`?

manikandan-thangavelu-rl replied:
Updated PR with the changes

SanketD92 commented on main/solution/prepare-devops-acc/config/infra/cloudformation.yml:
```diff
@@ -0,0 +1,33 @@
+                  - ec2:ModifyImageAttribute
+                  - ec2:DescribeImages
+                  - appstream:UpdateImagePermissions
+                Resource: "*"
```
Could we restrict this further, maybe by using a tag?

manikandan-thangavelu-rl replied:
Since the AMIs created using SWB Packer does not have any tags and also the user can use any custom AMI IDs (Like the ones created by RG repo also). We have not restricted this further.

manikandan-thangavelu-rl committed:
Merge remote-tracking branch 'upstream/feat-tre-enchancement-feature-1' into feat-tre-enchancement-feature-1

manikandan-thangavelu-rl committed:
Removed unwanted async, Added Default for AMI Sharing

narendran1560 committed:
Merge remote-tracking branch 'upstream/develop' into feat-tre-enhancements

narendran1560 committed:
Merge branch 'feat-tre-enhancements' of https://github.com/RLOpenCatalyst/service-workbench-on-aws into feat-tre-enhancements

narendran1560 committed:
Merge remote-tracking branch 'upstream/develop' into feat-tre-enchancement-feature-1

narendran1560 committed:
Merge branch 'feat-tre-enchancement-feature-1' of https://github.com/RLOpenCatalyst/service-workbench-on-aws into feat-tre-enchancement-feature-1

manikandan-thangavelu-rl committed:
Merge remote-tracking branch 'upstream/feat-tre-enchancement-feature-1' into feat-tre-enchancement-feature-1

narendran1560 committed:
Resolved PR comments which was rasied by Robert

narendran1560 committed:
Merge remote-tracking branch 'upstream/feat-tre-enhancements' into feat-feature-2and3

narendran1560 committed:
Merge pull request #73 from narendran1560/feat-feature-2and3

Resolved aws-swb PR comments

narendran1560 committed:
Merge pull request #74 from manikandan-thangavelu-rl/feat-tre-enchancement-feature-1

Chore: Incorporated PR Comments

narendran1560 committed:
Merge branch 'feat-tre-enchancement-integration' into feat-tre-enchancement-feature-1
